### PR TITLE
Add utilities and tests

### PR DESCRIPTION
**Pull request by soc221b**
I added some useful functions and tests for it.

The import error it occurs when I use those commands below:
1. `nosetests .`
2. `flake8 .`
3. `python3 tests/test.py`
4. `python3 file_ckecker/file_ckecker.py`
It seems that because folder name file_checker is same as file_checker.py in the folder, and loader is ambiguous.

### CODE REVIEW

File: src/file_checker.py [hunk 1]
```diff
@@ -29,17 +29,16 @@ def convert_sep(self, name, formats_):
...
-            for f in formats_:
-                if f == 'upper_case':
-                    name = name.upper()
-                if f == 'lower_case':
-                    name = name.lower()
-                if f == 'camel_case':
-                    pass
-                if f == 'pascal_case':
-                    pass
+            if format == 'upper_case':
```
Comment (arnavb) [inline src/file_checker.py]: I wouldn't use format as a name. It is a fairly common function that shouldn't cause any error here, but still.

File: tests/test.py [hunk 2]
```diff
@@ -1,36 +1,54 @@
...
 class TestUtil():
     def test_gen_unique_str(self):
         gen_unique_str = util.gen_unique_str
+        errors = []
         u = gen_unique_str('')
 
         s = ''
         if gen_unique_str(s) in s:
-            raise Exception("'{0}' is in '{1}'".format(gen_unique_str(s), s))
+            errors += ["'{0}' is in '{1}'".format(gen_unique_str(s), s)]
```
Comment (arnavb) [inline tests/test.py]: I think using the `.push` function would be clearer here.

File: tests/test.py [hunk 3]
```diff
@@ -1,36 +1,54 @@
...
         start = time.time()
         for i in range(0, 1000):
             # 256 chars
             gen_unique_str(u * 256)
 
-        assert time.time() - start < 0.5
+        if time.time() - start > 0.5:
+            errors += ['The algorithm is not efficient.']
+
+        if errors:
```
Comment (arnavb) [inline tests/test.py]: I personally don't like the approach of throwing an exception if there are errors. Testing usually involves a series of asserts (Which would print out the problem in the same format), and exceptions are meant for _exceptional_ behavior anyways. The only exception (Pun intended) I would make to this rule is for benchmarking. However, even then there is a [builtin module for profiling](https://docs.python.org/3.6/library/profile.html).

File: tests/test.py [hunk 4]
```diff
@@ -56,4 +83,258 @@ def test_get_first_word(self):
...
+        if time.time() - start > 5:
+            errors += ['The algorithm is not efficient.']
+
+        if errors:
+            raise Exception(get_error_string(errors))
+
+    def convert_words_to_case(self, words, case):
+        exp = []
+        if case == 'lower_case':
+            for w in words:
```
Comment (arnavb) [inline tests/test.py]: I believe you can use a list comprehension for these internal loops.

File: tests/test.py [hunk 5]
```diff
@@ -56,4 +83,258 @@ def test_get_first_word(self):
...
+        if case == 'upper_case':
+            for w in words:
+                exp += [w.upper()]
+        if case == 'camel_case' or case == 'pascal_case':
+            for w in words:
+                exp += [w.lower()]
+            for i, w in enumerate(exp):
+                if w:
+                    if len(w) > 1:
+                        w = w[0].upper() + w[1:]
```
Comment (arnavb) [inline tests/test.py]: Ternary possible here.

**Review by arnavb — CHANGES_REQUESTED** [review]

File: src/util.py [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+def gen_unique_str(string):
+    u = '_'
+    while u in string:
+        u += '_'
+    return u
+
+
+def get_first_word(s):
+    try:
+        beg = s.find(re.search('[A-Za-z]', s)[0]) if len(s) else -1
```
Comment (arnavb) [inline src/util.py]: Don't subscript. Use `.group(0)` to fix the bug. Otherwise you get the non-subscriptable exception thrown.

**Comment by arnavb** [comment]
@iattempt I was able to fix the grouping bug, but I am unsure how to fix the import problem you are getting, since the build is now passing.

**Comment by soc221b** [comment]
@arnavb I have fixed the import error, to reproduce on these error:
1. `git checkout -b 8365364`
2. I want to use class FileChecker, and I try to add these code to tests/test.py:

    1. import `module` and call FileChecker by module-name
    ```python
    import file_checker # noqa

    fc = file_checker.FileChecker()
    ```

    2. from `module` import `name`
    ```python
    from file_checker import FileChecker# noqa
    ```

3. `nosetests .`

output:
`ImportError: cannot import name 'FileChecker'` or
`AttributeError: module 'file_checker' has no attribute 'FileChecker'`

**Comment by arnavb** [comment]
@iattempt Alright, I'll look it over tomorrow!

**Review by arnavb — DISMISSED** [review]

**Comment by arnavb** [comment]
@iattempt I have found the solution to the import error. You are right, it is because of packages taking higher precedence over files. Since `file_checker` has an `__init__.py`, it takes higher precedence over the file inside. If you need to access the inner file, then follow the following steps:

1. Move all files in folder `src` to a new folder `file_checker`.
2. Change the respective line to:
    ```Python
    sys.path.append(os.sep.join([os.path.dirname(os.path.realpath(__file__)),
                             '..',
                             'file_checker'
                             ]))
    ```
3. Go to the topmost directory. (The one with `README.md`) and add an empty file `__init__.py`.
4. Run `nosetests . -v`

Output: 

```
test.TestFileChecker.test_convert_sep ... ok
This test is assumed that ... ok
test.TestUtil.test_gen_unique_str ... ok
test.TestUtil.test_get_first_word ... ok
test.TestUtil.test_get_words ... ok

----------------------------------------------------------------------
Ran 5 tests in 0.433s

OK
```

However, this is _not_ recommended practice, as it involves extra files. The better solution is to use an `src` model, which would work in this case and is the current approach.

I have the changes available in my current working directory (I have `git stash`ed them) and I can push them if desired.

Source: https://stackoverflow.com/a/4092424/6525260

**Comment by soc221b** [comment]
@arnavb Is this all right? I am going to merge this PR.

**Review by arnavb — APPROVED** [review]
Merge it! 😂

**Pull request merged**